frontend/src/pages/storageClasses/StorageClassEditModal.tsx

```diff
@@ -0,0 +1,129 @@
+  storageClass: StorageClassKind;
+  onSuccess: () => Promise<StorageClassKind[] | void>;
+  onClose: () => void;
+}
+
+export const StorageClassEditModal: React.FC<StorageClassEditModalProps> = ({
+  storageClass,
+  onSuccess,
+  onClose,
+}) => {
+  const { name: storageClassName } = storageClass.metadata;
+  const storageClassConfig = getStorageClassConfig(storageClass);
+  const [displayName, setDisplayName] = React.useState(storageClassConfig?.displayName ?? '');
+  const [description, setDescription] = React.useState(storageClassConfig?.description ?? '');
+  const [updateError, setUpdateError] = React.useState<Error>();
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+
+  const onSave = React.useCallback(async () => {
+    setIsSubmitting(true);
+
+    try {
+      await updateStorageClassConfig(storageClassName, { displayName, description });
+      await onSuccess();
+      onClose();
+    } catch (error) {
+      if (error instanceof Error) {
+        setUpdateError(error);
+      }
+    } finally {
+      setIsSubmitting(false);
+    }
+  }, [description, displayName, storageClassName, onClose, onSuccess]);
+
+  return (
+    <Modal
+      isOpen
+      variant={ModalVariant.small}
+      title="Edit storage class details"
+      onClose={onClose}
+      footer={
+        <DashboardModalFooter
+          onCancel={onClose}
+          onSubmit={onSave}
+          submitLabel="Save"
+          isSubmitLoading={isSubmitting}
+          isSubmitDisabled={!displayName || isSubmitting}
+          error={updateError}
+          alertTitle="Error updating storage class"
+        />
+      }
+      data-testid="edit-sc-modal"
+    >
+      <Alert
+        isInline
+        variant="info"
+        title="Editing these details will not affect the storage class in OpenShift."
+        className="pf-v5-u-mb-lg"
+      />
+
+      <Form id="edit-sc-form">
+        <DescriptionList>
+          <DescriptionListGroup>
+            <DescriptionListTerm>OpenShift storage class</DescriptionListTerm>
+            <DescriptionListDescription data-testid="edit-sc-openshift-class-name">
+              <Flex
+                spaceItems={{ default: 'spaceItemsSm' }}
+                alignItems={{ default: 'alignItemsCenter' }}
+              >
+                <FlexItem>{storageClassName}</FlexItem>
+                {isOpenshiftDefaultStorageClass(storageClass) && <OpenshiftDefaultLabel />}
+              </Flex>
+            </DescriptionListDescription>
+          </DescriptionListGroup>
+
+          <DescriptionListGroup>
+            <DescriptionListTerm>Provisioner</DescriptionListTerm>
+            <DescriptionListDescription data-testid="edit-sc-provisioner">
+              {storageClass.provisioner}
+            </DescriptionListDescription>
+          </DescriptionListGroup>
+        </DescriptionList>
+
+        <FormGroup label="Display name" isRequired fieldId="edit-sc-display-name">
+          <TextInput
```

Comment:
I'd suggestion to add an `id` equals to `fieldId` above here, so when you click on the label, it will also focus on the text input.